distribution/push_v2.go

```diff
@@ -118,12 +118,12 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, id
 		return fmt.Errorf("could not find image from tag %s: %v", reference.FamiliarString(ref), err)
 	}
 
-	rootfs, _, err := p.config.ImageStore.RootFSAndOSFromConfig(imgConfig)
+	rootfs, os, err := p.config.ImageStore.RootFSAndOSFromConfig(imgConfig)
 	if err != nil {
 		return fmt.Errorf("unable to get rootfs for image %s: %s", reference.FamiliarString(ref), err)
 	}
 
-	l, err := p.config.LayerStore.Get(rootfs.ChainID())
+	l, err := p.config.LayerStores[os].Get(rootfs.ChainID())
```

Comment:
`RootFS` could have a `Layer()` method for this. Or just replace `RootFSAndOSFromConfig`. (could be follow-up).

Reply:
Let's leave this for a follow-up for now unless someone else believes it's critical for this PR.